internal/resolve/catalog_test.go

```diff
@@ -727,3 +854,44 @@ func TestClusterExtensionNoMatchLabel(t *testing.T) {
 	require.Error(t, err)
 	require.ErrorContains(t, err, fmt.Sprintf("no package %q found", pkgName))
 }
+
+func TestUnequalPriority(t *testing.T) {
+	pkgName := randPkg()
+	w := staticCatalogWalker{
+		"a": func() (*declcfg.DeclarativeConfig, *catalogd.ClusterCatalogSpec, error) {
+			return genPackage(pkgName), &catalogd.ClusterCatalogSpec{Priority: 1}, nil
+		},
+		"b": func() (*declcfg.DeclarativeConfig, *catalogd.ClusterCatalogSpec, error) {
+			return genPackage(pkgName), &catalogd.ClusterCatalogSpec{Priority: 0}, nil
+		},
+	}
+	r := CatalogResolver{WalkCatalogsFunc: w.WalkCatalogs}
+
+	ce := buildFooClusterExtension(pkgName, "", ">=1.0.0 <=1.0.1", ocv1alpha1.UpgradeConstraintPolicyEnforce)
+	_, _, _, err := r.Resolve(context.Background(), ce, nil)
```

Comment:
I think we should ensure we actually got the right bundle here (from catalog `"a"`)

Reply:
Not sure we can actually check that out, since we don't return the catalog it comes from...

Reply:
What we've done in other tests is populated the different catalogs with different bundle versions, so i.e. catalog "a" has 1.0.0 and "b" has 1.0.1, so if you want the bundle from catalog a you can just check that the bundle version matches. Just make sure of course that the semver matches both (which it does now as written).

Reply:
Fair, I cheated and just used `genPackage()`